python/pyspark/ml/classification.py

```diff
@@ -3151,7 +3151,7 @@ def func(predictions):
                     predArray.append(x)
                 return Vectors.dense(predArray)
 
-            rawPredictionUDF = udf(func)
```

Comment:
Do I need to add a test here to ensure that the `rawPrediction` column is no longer `string`

https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/python/pyspark/ml/tests/test_algorithms.py#L108-L117

Reply:
Yeah, I think we should better add a test if possible.

Reply:
Got it, added a test

Reply:
@HyukjinKwon 
why only `transformed_df.head()` trigger this error ?
does it indicate bugs in pyspark-sql udf ?

Reply:
Seems like `pred.show()` triggers an exception too? what does it return in other methods?